Code/Core/FileIO/FileIO.cpp

```diff
@@ -222,6 +223,20 @@
             return false;
         }
         linkPath.SetLength( length );
+
+        if (linkPath.Get()[0] != '/' )
+        {
+            // Convert linkPath to a full path in case the relative path
+            // is wrong from dstFileName
+            if ( srcFileName[0] != '/' )
+            {
+                return false;
+            }
+            const char *pLastSlash = strrchr(srcFileName, '/');
+            const size_t lenSrcPath = (pLastSlash - srcFileName) + 1;
+            linkPath.Format("%.*s%s", (int)lenSrcPath, srcFileName, linkPath.Get());
```

Comment:
`%.*s` is a bit obscure, is it ok to use?

Reply:
This is probably fine, but it might be better to use Append() to construct the string. It'll certainly be more performant if nothing else (not that this is of particular concern here)